indexer/src/main/kotlin/com/openlattice/BackgroundExternalDatabaseSyncingService.kt

```diff
@@ -209,27 +209,37 @@ class BackgroundExternalDatabaseSyncingService(
         }.toSet()
     }
 
-    private fun removeNonexistentTablesAndColumnsForOrg(orgId: UUID, existingTableIds: Set<UUID>, existingColumnIds: Set<UUID>) {
+    private fun removeNonexistentTablesAndColumnsForOrg(
+        orgId: UUID,
+        existingTableIds: Set<UUID>,
+        existingColumnIds: Set<UUID>
+    ) {
 
         // delete missing tables
 
-        val tableIdsToDelete = organizationExternalDatabaseTables.keySet(Predicates.equal(ORGANIZATION_ID_INDEX, orgId)).filter {
-            !existingTableIds.contains(it)
-        }.toSet()
+        val tablesToDelete: Set<OrganizationExternalDatabaseTable> = organizationExternalDatabaseTables
+            .values(Predicates.equal(ORGANIZATION_ID_INDEX, orgId))
+            .filter { !existingTableIds.contains(it.id) }
+            .toSet()
+        val tableIdsToDelete = tablesToDelete.mapTo(mutableSetOf<UUID>()) { it.id }
 
         if (tableIdsToDelete.isNotEmpty()) {
             edms.deleteOrganizationExternalDatabaseTableObjects(tableIdsToDelete)
+            organizationMetadataEntitySetsService.deleteDatasets(orgId, tablesToDelete)
```

Comment:
imo i think it would be a bit cleaner/safer to add this logic as part of the existing delete external tables workflow (within ExternalDatabaseManagementService) but it doesn't really matter

Reply:
I thought about it, but decided against it because I wanted to avoid making changes that will have to be undone since the plan is to nuke `OrganizationMetadataEntitySetsService` anyway.